File: src/main/java/no/nav/fo/veilarbregistrering/oppgave/OppgavePredicates.java
package no.nav.fo.veilarbregistrering.oppgave;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.function.Predicate;

public class OppgavePredicates {

    public static Predicate<OppgaveImpl> oppgaveAvTypeOppholdstillatelse() {
        return o -> o.getOppgavetype().equals(OppgaveType.OPPHOLDSTILLATELSE);
    }

    public static Predicate<OppgaveImpl> oppgaveOpprettetForMindreEnnToArbeidsdagerSiden(LocalDate dagensDato) {
        return o -> o.getOpprettet().erMindreEnnToArbeidsdagerSiden(dagensDato);
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/oppgave/OppgavePredicates.kt
package no.nav.fo.veilarbregistrering.oppgave

import java.time.LocalDate
import java.util.function.Predicate

object OppgavePredicates {
    fun oppgaveAvType(oppgaveType: OppgaveType): Predicate<OppgaveImpl> {
        return Predicate { o: OppgaveImpl -> o.oppgavetype == oppgaveType }
    }

    fun oppgaveOpprettetForMindreEnnToArbeidsdagerSiden(dagensDato: LocalDate): Predicate<OppgaveImpl> {
        return Predicate { o: OppgaveImpl -> o.opprettet.erMindreEnnToArbeidsdagerSiden(dagensDato) }
    }
}